tiledb/sm/query/query.cc

```diff
@@ -2093,8 +2096,8 @@ void Query::copy_aggregates_data_to_user_buffer() {
   }
 }
 
-RestClient* Query::rest_client() const {
-  return storage_manager_->rest_client();
+shared_ptr<RestClient> Query::rest_client() const {
```

Comment:
`shared_ptr` seems like excessive overhead here. Since the `StorageManager` implementation was calling `get()` to obtain a pointer, that'll be fine here as well.

The existence of this function at all is somewhat questionable, but that's outside the scope of this PR.

Reply:
It's used _solely_ in the `QueryPlan` constructor. We could eventually remove this function and pass a `ContextResources&` into that constructor instead. 
`ContextResources` returns a `shared_ptr<RestClient>` so I changed to reflect that class upon the removal of SM but can revert back. I have no preference.

Reply:
Let's get it in now rather than delay for another review round. 

>It's used solely in the QueryPlan constructor. We could eventually remove this function and pass a ContextResources& into that constructor instead.

That's the right way to proceed.